x-pack/legacy/plugins/infra/public/pages/logs/analysis/setup/initial_configuration_step/analysis_setup_timerange_form.tsx

```diff
@@ -105,27 +111,44 @@ export const AnalysisSetupTimerangeForm: React.FunctionComponent<{
           <EuiFormControlLayout
             clear={endTime ? { onClick: () => setEndTime(undefined) } : undefined}
           >
-            <EuiDatePicker
-              showTimeSelect
-              selected={endTimeValue}
-              onChange={date => setEndTime(selectedDateToParam(date))}
-              placeholder={endTimeDefaultDescription}
-              openToDate={now}
-              minDate={startTimeValue}
-              minTime={
-                selectedEndTimeIsToday
-                  ? now
-                  : moment()
-                      .hour(0)
-                      .minutes(0)
-              }
-              maxTime={moment()
-                .hour(23)
-                .minutes(59)}
-            />
+            <WithFixedDatepickerZIndex>
+              {popperClassName => (
+                <EuiDatePicker
+                  showTimeSelect
+                  selected={endTimeValue}
+                  onChange={date => setEndTime(selectedDateToParam(date))}
+                  placeholder={endTimeDefaultDescription}
+                  openToDate={now}
+                  minDate={startTimeValue}
+                  minTime={
+                    selectedEndTimeIsToday
+                      ? now
+                      : moment()
+                          .hour(0)
+                          .minutes(0)
+                  }
+                  maxTime={moment()
+                    .hour(23)
+                    .minutes(59)}
+                  popperClassName={popperClassName}
+                />
+              )}
+            </WithFixedDatepickerZIndex>
           </EuiFormControlLayout>
         </EuiFlexGroup>
       </EuiFormRow>
     </EuiDescribedFormGroup>
   );
 };
+
+const WithFixedDatepickerZIndex = euiStyled(
```

Comment:
Maybe save some complexity by doing:
```suggestion
const StyledDatePicker = euiStyled(
  ({
    className,
    ...rest,
  }: {
    className?: string;
    rest: Object; // Not sure off the top of my head how to refer to EuiDatePicker's propTypes
  }) => <EuiDatePicker {...rest} popperClassName={className}/>
)`
  z-index: 3 !important;
`;
```

Reply:
Ok, I can do something to that effect. The reason why I implemented it as a FAAC component was to avoid that coupling to the datepicker props and retain the ability to supply a `className` to the datepicker.